webhooks-extension/src/components/WebhookCreate/WebhookCreate.js

```diff
@@ -0,0 +1,698 @@
+          this.setState({
+            namespaceFail: true,
+            notificationErrorMessage: "Failed to get namespaces, error returned was : " + text,
+            notificationStatus: 'error',
+            notificationStatusMsgShort: 'Error:',
+            showNotification: true,
+          });
+        });
+    }
+  }
+
+  async fetchPipelines(namespace) {
+    let pl;
+    try {
+      pl = await getPipelines(namespace);
+      this.setState({apiPipelines: pl})
+    } catch (error) {
+        error.response.text().then((text) => {
+          this.setState({
+            pipelineFail: true,
+            notificationErrorMessage: "Failed to get pipelines, error returned was : " + text,
+            notificationStatus: 'error',
+            notificationStatusMsgShort: 'Error:',
+            showNotification: true,
+          });
+        });
+    }
+  }
+  
+  async fetchSecrets(namespace) {
+    let s;
+    try {
+      s = await getSecrets(namespace);
+      this.setState({apiSecrets: s})
+    } catch (error) {
+        error.response.text().then((text) => {
+          this.setState({
+            secretsFail: true,
+            notificationErrorMessage: "Failed to get secrets, error returned was : " + text,
+            notificationStatus: 'error',
+            notificationStatusMsgShort: 'Error:',
+            showNotification: true,
+          });
+        });
+    }
+  }
+
+  async fetchServiceAccounts(namespace) {
+    let sa;
+    try {
+      sa = await getServiceAccounts(namespace);
+      this.setState({apiServiceAccounts: sa})
+    } catch (error) {
+        error.response.text().then((text) => {
+          this.setState({
+            serviceAccountsFail: true,
+            notificationErrorMessage: "Failed to get service accounts, error returned was : " + text,
+            notificationStatus: 'error',
+            notificationStatusMsgShort: 'Error:',
+            showNotification: true,
+          });
+        });
+    }
+  }
+
+  handleChange = (event) => {
+    const target = event.target;
+    const value = target.value;
+    const name = target.name;
+    this.setState({[name]: value});
+  }
+
+  handleChangeNamespace = (itemText, value) => {
+    this.setState({
+      namespace: itemText.selectedItem,
+      apiPipelines: '',
+      apiSecrets: '',
+      apiServiceAccounts: '',
+      pipeline: '',
+      gitsecret: '',
+      serviceAccount: '',
+    });
+    if (!this.state.pipelineFail) {
+      this.fetchPipelines(itemText.selectedItem);
+    }
+    this.fetchSecrets(itemText.selectedItem);
+    this.fetchServiceAccounts(itemText.selectedItem);
+  }
+
+  handleChangePipeline = (itemText, value) => {
+    this.setState({pipeline: itemText.selectedItem });
+  }
+
+  handleChangeSecret = (itemText, value) => {
+    this.setState({gitsecret: itemText.selectedItem });
+  }
+
+  handleChangeServiceAcct = (itemText, value) => {
+    this.setState({serviceAccount: itemText.selectedItem });
+  }
```

Comment:
You should be able to bring these three methods into one method by just substituting the state name, gitsecret etc, also value is not used here